### DISCUSSION TITLE
Preserve the original color for carbon elements in 'element-symbol' theme

### DISCUSSION THREAD

**Discussion by Marwa-Da** [molstar/molstar#1034] [Q&A] [answered]
Hi everyone,

Recently, I've been exploring different strategies to preserve the original color of carbon elements when implementing the `element-symbol` theme but couldn't manage to do it. So I'm looking for some ideas.

The goal is to apply a specific coloring theme then applying the **element-symbol** and leaving the carbon color unaltered.

I tried to get the current theme applied on the node and send it to the `carbonColor.name`  parameter but it fails in case of applying the `element-symbol` theme twice in a row as in the second round the theme will be `element-symbol` for sure :(

**Comment by Marwa-Da** [comment]
@dsehnal Any ideas, please? Still need help

**Reply by dsehnal** [reply]
What you want is the carbons to remain gray?

**Reply by dsehnal** [reply]
`carbonColor: { name: 'element-symbol', params: {} }` doesn't work?

**Reply by Marwa-Da** [reply]
I don't want to color the carbons so it keep its old color unaltered. If we applied rainbow scheme then we applied element-symbol, the carbons must preserve the color it got from applying rainbow scheme.

**Reply by dsehnal** [reply]
There is only limited themes support (see [here](https://github.com/molstar/molstar/blob/a1e2b0e5aec4d92fe84fcf8e6bfe1f08cdc08a29/src/mol-theme/color/element-symbol.ts#L69)) for the carbon color. If you want something more custom, you will likely have to write your own version of the `element-symbol` theme.

**Reply by Marwa-Da** [reply]
That is correct and I was editing the `element-symbol` theme but got stuck on how to get the color of the carbon atom at hand while only having `location: Location` in the `color` method.

**Reply by dsehnal** [reply]
You need to create an instance of the secondary theme inside your custom one and call the `.color` function similar to this (notice the `?.color`):

```ts
    const carbonColor = getCarbonTheme(ctx, props.carbonColor)?.color;

    function elementColor(element: ElementSymbol, location: Location) {
        return (carbonColor && element === 'C')
            ? carbonColor(location, false)
            : elementSymbolColor(colorMap, element);
    }
```

**Reply by Marwa-Da** [reply]
You mean I need to pass the secondary theme (old theme) to `element-symbol` so we can apply it inside, right?

**Reply by dsehnal** [reply, marked as answer]
Yes, but you will likely have to write a custom version of element-symbol theme for this to work as you expect.

**Reply by Marwa-Da** [reply]
Really appreciate your help, thank you!
